units/system-systemd\x2dcryptsetup.slice

```diff
@@ -0,0 +1,13 @@
+#  SPDX-License-Identifier: LGPL-2.1+
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
```

Comment:
Drive-by question. I'm a bit confused about this file. As far as I see it is unused. What is the purpose of this slice unit?

Reply:
@fbuihuu ?

Reply:
all instance units of the same template implicitly get their own slice assigned covering all instances. This unit file just attaches a nice description to that slice unit, which is a unit that is synthesized anyway, but otherwise wouldn't have a nice description

It might make sense we carefully makes sure we do the same for all such implicit slices synthesized for multi-instantiated units.

Reply:
It's used automatically by `systemd-cryptsetup@.service`. But the commit doesn't say why this is needed, and even though the unit file refers to systemd.special(7), the unit is not mentioned there. I'm not sure if this was unit was left over from an earlier version of the patchset, or if actually serves a purpose.

Reply:
So... is the Description line the only purpose? I'd just drop it then, it doesn't seem worth it, and we don't do it for other units.

Reply:
According to the commit message: "Note that the slice containing the cryptsetup services also needs to drop the
usual shutdown dependencies as it's required by the cryptsetup services."

So the main purpose is to drop the usual shutdown deps I would say. Maybe PID1 got the ability to do that automatically now ?

Reply:
Oh, so the slice indeed does have `DefaultDependencies=no`.
So it seems all good, except that we should document the slice in systemd.special(7) or remove the Documentation link.

Reply:
https://github.com/systemd/systemd/pull/19260